api/acl.go

```diff
@@ -212,11 +212,15 @@ type ACLPolicyListStub struct {
 
 // ACLPolicy is used to represent an ACL policy
 type ACLPolicy struct {
-	Name        string
-	Description string
-	Rules       string
-	CreateIndex uint64
-	ModifyIndex uint64
+	Name         string
+	Description  string
+	Rules        string
+	JobNamespace string
+	JobID        string
+	Group        string
```

Comment:
Nitpicking but having `Namespace` and `ID` prefixed with `Job` and `Group` and `Task` not feels kinda weird.

Reply:
I like that, it makes it obvious that this is a special use (and has a few words fewer memory allocations for the common case).

Reply:
Now I'm remembering why I didn't do this originally, which is that the state store index schema was super weird. Lemme spend a little bit to see if I can figure out how to do it. Indexing on `"JobACL.Namespace"` actually does the right thing, but only if `JobACL` is non-nil so it'll break everywhere else.

Reply:
Just needed a custom indexer. Done.